addons/stock/views/stock_quant_views.xml

```diff
@@ -148,6 +148,16 @@
         <field name="name">Current Stock</field>
         <field name="res_model">stock.quant</field>
     </record>
+    <record model="ir.actions.act_window" id="category_open_putaway"> <!-- Putaway rules from category -->
+        <field name="name">Putaway Rules</field>
+        <field name="res_model">product.putaway</field>
+        <field name="context">{'search_default_category_id': [active_id]}</field>
+    </record>
+    <record model="ir.actions.act_window" id="location_open_putaway"> <!-- Putaway rules from location -->
+        <field name="name">Putaway Rules</field>
+        <field name="res_model">product.putaway</field>
+        <field name="context">{'search_default_location_in_id': [active_id]}</field>
+    </record>
```

Comment:
are you sure you're making these changes in the right file? stock_quant_views.xml